apps/aehttp/src/sc_ws_handler.erl

```diff
@@ -22,6 +22,7 @@
 
 init(Req, _Opts) ->
     lager:debug("init(~p, ~p)", [Req, _Opts]),
+    process_flag(trap_exit, true),
```

Comment:
Based on the [cowboy doc](https://ninenines.eu/docs/en/cowboy/2.2/guide/handlers/#_cleaning_up), this is not needed for the terminate to be called. Could you please elaborate on why it addresses?

My concern is unintended behaviour change (in calling process) hence distinct behaviour in error case in tests and user experience.

Reply:
This is to ensure that the ws handler is killed immediately through cascading exit before being able to process the termination info message sent by the fsm. That is, the fsm sends an info msg before terminating, but the exit may propagate to the linked ws handler, killing it, before it has time to receive and forward that message. Trapping exits ensures that it will be able to process all messages sent by the fsm before it gets to the `'DOWN'` message.